util.go

```diff
@@ -28,12 +28,18 @@ func Join(ms ...Multiaddr) Multiaddr {
 
 	length := 0
 	for _, m := range ms {
+		if m == nil {
+			continue
+		}
 		length += len(m.Bytes())
 	}
 
 	bidx := 0
 	b := make([]byte, length)
 	for _, mb := range ms {
+		if mb == nil {
+			continue
+		}
 		bidx += copy(b[bidx:], mb.Bytes())
 	}
 	return &multiaddr{bytes: b}
```

Comment:
This will return a multiaddr with empty bytes when passed all nils. Should we have a special check for that? This seems okay, but I'm not sure how this empty multiaddress will behave everywhere else.

Reply:
Should we Fuzz test calling `String()`, `Protocols()` etc on the results of `SplitFirst` and `SplitLast?`